bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,251 @@
+                                                new DateTimeType(pet.status.activity.getLocationChanged()));
+                                    } catch (NumberFormatException e) {
+                                        logger.warn("Invalid location id: {}, ignoring command", newLocationIdStr);
+                                    } catch (SurePetcareApiException e) {
+                                        logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
+                                                newLocationIdStr, pet);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case PET_CHANNEL_LOCATION_TIMEOFFSET:
+                    logger.debug("Received location time offset update command: {}", command.toFullString());
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcarePet pet = petcareAPI.getPet(thing.getUID().getId());
+                            if (pet != null) {
+                                String commandIdStr = ((StringType) command).toString();
+                                try {
+                                    Integer commandId = Integer.valueOf(commandIdStr);
+                                    Integer currentLocation = pet.status.activity.where;
+                                    logger.debug("Received new location: {}", currentLocation == 1 ? 2 : 1);
+                                    // We set the location to the opposite state.
+                                    // We also set location to INSIDE (1) if currentLocation is Unknown (0)
+                                    if (commandId == 10 || commandId == 30 || commandId == 60) {
+                                        petcareAPI.setPetLocation(pet, currentLocation == 1 ? 2 : 1, new Date(
+                                                System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(commandId)));
+                                    }
+                                    updateState(PET_CHANNEL_LOCATION,
+                                            new StringType(pet.status.activity.where.toString()));
+                                    updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                            new DateTimeType(pet.status.activity.getLocationChanged()));
+                                    updateState(PET_CHANNEL_LOCATION_TIMEOFFSET, UnDefType.UNDEF);
+                                } catch (NumberFormatException e) {
+                                    logger.warn("Invalid location id: {}, ignoring command", commandIdStr);
+                                } catch (SurePetcareApiException e) {
+                                    logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
```

Comment:
Consider logging the stacktrace, maybe on debug? IO related exceptions happen when communicating with online services and are worth a warn log, see also: 

> to log exceptions in case of temporary problems, like connection problems. In case of such exceptions this should be reflected in an updated state of the binding
[[source](https://www.openhab.org/docs/developer/guidelines.html#f-logging)]

Should you update the thing status?

Reply:
Please apply this also to the methods above.

Reply:
hmmm... good point... i*ll think about this and do some tests